docs/pages/kubernetes-access/guides/dynamic-registration.mdx

```diff
@@ -0,0 +1,514 @@
+---
+
+With dynamic Kubernetes cluster registration, you can manage the Kubernetes
+clusters connected to your Teleport cluster without needing to modify the
+configuration file of an individual Kubernetes Service instance. 
+
+Dynamic Kubernetes cluster registration is useful when you have deployed
+multiple Kubernetes Service instances or need to regularly reconfigure access to
+Kubernetes clusters in your instrastructure.
+
+In this guide, we will show you how to set up dynamic Kubernetes cluster
+registration, then create, list, update, and delete Kubernetes clusters via
+`tctl`.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A Linux host where you will install the Teleport Kubernetes Service.
+
+  <Notice type="tip">
+
+  Our `teleport-kube-agent` Helm chart does not support dynamic Kubernetes
+  cluster registration.
+
+  </Notice>
+
+- A Kubernetes cluster to join to your Teleport cluster. You must have
+  permissions to create namespaces, secrets, service accounts, cluster roles,
+  and cluster role bindings in the cluster.
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/3. Set up the Teleport Kubernetes Service
+
+The Teleport Kubernetes Service proxies traffic from Teleport users to a
+Kubernetes API server so you can take advantage of passwordless authentication,
+role-based access controls, audit logging, and other Teleport features in order
+to manage access to Kubernetes. 
+
+In this step, you will install the Teleport Kubernetes Service on a Linux host
+and configure it to access any Kubernetes cluster you register with your
+Teleport cluster. 
+
+### Get a join token
+
+Establish trust between your Teleport cluster and your new Kubernetes Service
+instance by creating a join token:
+
+```code
+$ tctl nodes add --roles=kube
+The invite token: (=presets.tokens.first=)
+This token will expire in 30 minutes.
+
+Run this on the new node to join the cluster:
+
+> teleport start \
+   --roles=kube \
+   --token=(=presets.tokens.first=) \
+   --ca-pin=(=presets.ca_pin=) \
+   --auth-server=192.0.2.255:3025
+
+Please note:
+
+  - This invitation token will expire in 30 minutes
+  - 192.0.2.255:3025 must be reachable from the new node
+```
+
+Copy the token and keep it somewhere safe so you can use it when running the
+Teleport Kubernetes Service.
+
+### Install the Teleport Kubernetes Service
+
+Install the Teleport Kubernetes Service on your Linux host:
+
+(!docs/pages/includes/install-linux.mdx!)
+
+### Configure the Teleport Kubernetes Service
+
+On the host where you will run the Teleport Kubernetes Service, run the
+following command to create a base configuration for your Teleport instance,
+assigning `PROXY_SERVICE` to the host and port of your Teleport Proxy Service or
+Teleport Cloud tenant and `TOKEN` to the join token we created earlier:
+
+```code
+$ PROXY_SERVICE=teleport.example.com:443;
+$ TOKEN=(=presets.tokens.first=);
```

Comment:
I had a different take on using shell vars like this in comments below that I made earlier. Now that we have the in-page edits feature documented, could/should we use them instead?

Reply:
I've used in-page edits, but also kept the environment variables so we can include comments above them with a suggested value.